java/src/jmri/Block.java

```diff
@@ -208,27 +279,40 @@ public Sensor getSensor() {
         return _namedSensor;
     }
 
+    /**
+     * Property name change fired when a Sensor is set to / removed from a Block.
+     * The fired event includes
+     * old value: Sensor Bean Object if previously set, else null
+     * new value: Sensor Bean Object if being set, may be null if Sensor removed.
+     */
+    public final static String BLOCK_REPORTER_CHANGE = "BlockReporterChange"; // NOI18N
+    
     /**
      * Set the Reporter that should provide the data value for this block.
-     *
+     * Fires propertyChange "BlockReporterChange" when changed.
      * @see Reporter
      * @param reporter Reporter object to link, or null to clear
      */
     public void setReporter(Reporter reporter) {
+        if (Objects.equals(reporter,_reporter)) {
+            return;
+        }
         if (_reporter != null) {
             // remove reporter listener
             if (_reporterListener != null) {
                 _reporter.removePropertyChangeListener(_reporterListener);
                 _reporterListener = null;
             }
         }
+        Reporter oldReporter = _reporter;
         _reporter = reporter;
         if (_reporter != null) {
             // attach listener
             _reporter.addPropertyChangeListener(_reporterListener = (PropertyChangeEvent e) -> {
                 handleReporterChange(e);
             });
         }
+        firePropertyChange(BLOCK_REPORTER_CHANGE, oldReporter, reporter);
```

Comment:
I suggest that you check if oldReporter != reporter. If not, there is no change.

Reply:
Probably not easy to spot with this amount of edits in a file, however at the start of the method, there's a new check for this,
`if (Objects.equals(reporter,_reporter)) {
    return;
}`
which covers the scenario, unit tests for this also.
In future would be easier to read if I had copied the local oldSensor earlier in the method . . .
If latest commit doesn't pass unit tests I'll tweak.

Reply:
Thanks! I missed that.